PublicMapping/districtbuilder
# Highlight selected/hovered districts and zoom to district

## Overview
Highlight selected/hovered districts and zoom to district.

### Checklist


### Demo
![db_district_borders_zoom](https://user-images.githubusercontent.com/2926237/114583343-7ee1aa80-9c4f-11eb-8cb0-845eec526643.gif)

## Notes
There are two distinct pieces of functionality here:
1. District border added on select/hover
2. Zoom to district

For the border, I ended up reusing the `outlineColor` property which is already used for the "Find Unassigned" feature. By always setting the outline color via logic in TS (and to "transparent" if nothing should be shown) we no longer need the effect to conditionally hide that layer based on `findOpen`. Another change related to borders is moving the state capturing if a sidebar row is hovered up into Redux state since that is now relevant to the map as well, which allowed the component-level state to be deleted from `SidebarRow`.

For the zoom to district functionality, that reuses existing menu styles, borrow heavily from `ProjectListFlyout`, which overriding a few things to make the menu appear to the right of the sidebar. One other thing to note there is that there is currently only one item in the list but in the future there will be multiple items (as we add more items `MenuKeys` is meant to become a union type). 

@jfrankl The styles on the district options flyout look pretty close to me but maybe not to you :sweat_smile: Let me know if you think it's good enough for now.

## Testing Instructions
* Make sure selecting a district adds a border around the district of the same color as the district
* Make sure hovering a district row in the sidebar adds a border around the district of the same color as the district
* Make sure you can zoom to a district that isn't empty
* Make sure that zooming to an empty district (no geounits assigned) does nothing

Closes #612
